tests/h/search/query_test.py

```diff
@@ -165,6 +165,81 @@ def search(self, search, pyramid_request):
 
 
 class TestAnyMatcher(object):
+    def test_matches_uriparts(self, search, Annotation):
+        Annotation(target_uri="bar.com")
```

Comment:
As with the URI filters tests, I suggest tests should use valid absolute URIs for `target_uri`, unless it is specifically a test for what happens with invalid URIs, which I think will be covered by the `UriFilter` tests.